### PR TITLE
fix vbft shard user stake and seal block bug

### PR DESCRIPTION
**Pull request by qiluge**
1. fix stake param shard id serialize bug;
2. update vbft parent height at each proposal;
3. only update user unbound ong info while stake asset is ONT contract

### CODE REVIEW

File: smartcontract/service/native/shard_stake/params.go [hunk 1]
```diff
@@ -189,12 +204,12 @@ func (this *UnfreezeFromShardParam) Deserialize(r io.Reader) error {
 }
 
 type WithdrawStakeAssetParam struct {
-	ShardId types.ShardID
+	ShardId uint64
```
Comment (laizy) [inline smartcontract/service/native/shard_stake/params.go]: why modify this?

File: smartcontract/service/native/shard_stake/params.go [hunk 2]
```diff
@@ -208,7 +223,7 @@ func (this *WithdrawStakeAssetParam) Deserialize(r io.Reader) error {
 	if err != nil {
 		return fmt.Errorf("deserialize: read shard id failed, err: %s", err)
 	}
-	this.ShardId, err = types.NewShardID(id)
+	this.ShardId = id
```
Comment (laizy) [inline smartcontract/service/native/shard_stake/params.go:226]: not all uint64 are valid ShardID, the original check is needed.

**Review by laizy — CHANGES_REQUESTED** [review]

File: smartcontract/service/native/shard_stake/params.go [hunk 3]
```diff
@@ -421,14 +463,14 @@ func (this *UserStakeParam) Deserialize(r io.Reader) error {
 }
 
 type ChangeMaxAuthorizationParam struct {
-	ShardId    types.ShardID
+	ShardId    uint64
```
Comment (laizy) [inline smartcontract/service/native/shard_stake/params.go]: revert all those changes back

**Comment by qiluge** [comment]
@laizy fixed shard serialize problem

**Review by Honglei-Cong — APPROVED** [review]

**Pull request merged**